ci/main.go

```diff
@@ -0,0 +1,211 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"dagger.io/dagger"
+)
+
+const (
+	API_VERSION = "v1"
+
+	Go2WASM   = "go1.21"
+	GO2HOST   = "go1.20.4"
+	GO2PLUGIN = "go1.20.4"
+
+	// EXTRA ARGS FOR BUILDING (placed after the "go build")
+	// use -x for more details from a go compiler
+	EXTRA_HOST_ARGS = ""
+
+	SYSCALL_CLIENT_SIDE = "apiwasm/syscall/*.go"
+
+	REP = "g/file/" + API_VERSION + "/file.pb.go"
+)
+
+var (
+	// protobuf files
+	apiProto  string
+	testProto string
+
+	// protoc plugin
+	template    string
+	eneratorSrc string
+)
+
+func main() {
+	if err := build(context.Background()); err != nil {
+		log.Fatalf("Cannot build dagger pipeline %v\n", err)
+	}
+}
+
+func build(ctx context.Context) error {
+	fmt.Println("Building with Dagger")
+
+	// initialize Dagger client
+	// set the current(root) directory on the host as the working directory
+	client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	// get reference to the local Dockerfile
+	dockerDir := client.Host().Directory("./ci/")
+
+	// initialize image fron dockerfile
+	// mount a host directory in the container at the '/workspaces/parigot' path
+	img := client.Container().Build(dockerDir).WithDirectory("/workspaces/parigot", client.Host().Directory("."),
+		dagger.ContainerWithDirectoryOpts{Exclude: []string{".devcontainer/", "multibuild/", "build/"}}).WithWorkdir("/workspaces/parigot")
+
+	// // get version execute
```

Comment:
If these comments are no longer needed, please delete them.

The rule I try to use is 
> a ***WRONG***  comment is worse than no comment

Related:
> Trust the tree.

The latter one is about trusting that you have all the old versions committed in the repository, so there is no use in keeping things around "because you might need it later."  I don't always get this right and feel free to fix it when you see I've done it.